docs/testing-operators.md

```diff
@@ -1,120 +1,404 @@
+git clone https://github.com/operator-framework/operator-lifecycle-manager.git
+```
+
+Before you begin your current working dir should look like the following, with `my-operator` as an example for the name of your bundle:
 
-`test-operator` injects a scorecard proxy container and volume into an operators' CSV manifest before deployment; this is necessary to get API server logs, from which the scorecard determines runtime test results. These modifications are not persistent, as they're only needed for testing.
+```
+my-operator
+operator-marketplace
+operator-courier
+operator-lifecycle-manager
+```
 
-**Note**: no explicit number of points or percentage is necessary to achieve before merging _yet_. These are suggestions to improve your operator.
+### Tools
 
 #### operator-courier
 
-The [`operator-courier verify`][courier] command verifies that a set of files is valid and can be bundled and pushed to [quay.io][quay]. Read the [docs][courier-docs] for more information.
+`operator-courier` is used for metadata syntax checking and validation. This can be installed directly from `pip`:
+
+```
+pip3 install operator-courier
+```
+
+#### Quay Login
+
+In order to test the Operator installation flow, store your Operator bundle on [quay.io](https://quay.io). You can easily create an account and use the free tier (public repositories only). To upload your Operator to quay.io a token is needed. This only needs to be done once and can be saved locally. The `operator-courier` repository has a script to retrieve the token:
+
+```
+./operator-courier/scripts/get-quay-token
+
+Username: johndoe
+Password: 
+{"token": "basic abcdefghijkl=="}
+```
+
+A token takes the following form and should be saved in an environment variable:
+
+```
+export QUAY_TOKEN="basic abcdefghijkl=="
+```
+### Linting
+
+`operator-courier` will verify the fields included in the Operator metadata (CSV). The fields can also be manually reviewed according to [the operator CSV documentation](https://github.com/operator-framework/community-operators/blob/master/docs/required-fields.md).
+
+The following command will run `operator-courier` against the bundle directory `my-operator/` from the example above.
+
+```
+operator-courier verify --ui_validate_io my-operator/
+```
+
+If there is no output, the bundle passed `operator-courier` validation. If there are errors, your bundle will not work. If there are warning we still encourage you to fix them before proceeding to the next step.
+
+### Push to quay.io
+
+The Operator metadata in it's bundle format will be uploaded into your namespace in [quay.io](http://quay.io).
+
+> The value for `PACKAGE_NAME` *must* be the same as in the operator's `*package.yaml` file and the operator bundle directory name. Assuming it is `my-operator`, this can be found by running `cat my-operator/*.package.yaml`.
 
-### Upstream operators
+> The `PACKAGE_VERSION` is entirely up for you to decide. Best practice is it coincides with your Operator version.
```

Comment:
This best practice is very hard to maintain.

Reply:
+1, it's not generally possible to keep these in sync.

Reply:
What's the suggestion we should give here?

Reply:
Perhaps we should describe how it works by explaining that, since each package is built to contain every version, the package version is independent of your operator's latest version.

The reality is that this version is a weird artifact of the fact that this is a hacky implementation of app registry built on top of a thing that stores helm charts in quay, so the real answer is: this version number doesn't matter at all, you just need to increment it every time since it is otherwise irrelevant and orthogonal to the goal of versioning your operator.